src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.Unix.cs

```diff
@@ -31,8 +32,31 @@ internal static string ParseFileName(string path, ZipVersionMadeByPlatform madeB
             }
             else
             {
+                PathInternal.CheckInvalidPathChars(path);
                 return Path.GetFileName(path);
             }
         }
+
+        internal static readonly char[] InvalidPathChars =
```

Comment:
I thought not because it would interfere with the current-OS call to PathInternal.CheckInvalidPathChars on line 35. If PathInternal.CheckInvalidChars.OSGroup is partial as part of PathInternal then we can't use more than one of them, but if we're willing to do a bit more refactoring then we could make it work. 

Options I can think of:
- We could leave PathInternal.CheckInvalidChars the way it is (as part of partial class PathInternal) but add PathInternal.CheckInvalidCharsWindows/Unix functions.
- Add a CheckInvalidChars overload that takes a PlatformID.
- Add non-partial classes PathInternal_Windows/Unix that contains CheckInvalidChars functions. This wouldn't interfere with regular use of the PathInternal partial class but would allow ```PathInternal_Windows.CheckInvaidChars``` to work on Unix.

Am I missing a better option?